waku/waku_sync/session.nim

```diff
@@ -0,0 +1,142 @@
+when (NimMajor, NimMinor) < (1, 4):
+  {.push raises: [Defect].}
+else:
+  {.push raises: [].}
+
+import std/options, stew/results, chronicles, chronos, libp2p/stream/connection
+
+import ../common/nimchronos, ../waku_core, ./raw_bindings, ./storage_manager
+
+logScope:
+  topics = "waku sync"
+
+type SyncSessionType* = enum
+  CLIENT = 1
+  SERVER = 2
```

Comment:
Do we expect more session type in the future?

Reply:
I don't think so